vllm/spec_decode/multi_step_worker.py

```diff
@@ -24,14 +29,67 @@ class MultiStepWorker(Worker, ProposerWorkerBase):
     requires more thought for MultiStepWorker support.
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, draft_ranks: Optional[List[int]] = None, **kwargs):
+        """Create a MultiStepWorker.
+
+        It allows a speculative draft model to run with smaller tensor
+        parallel degree than target model.
+        This reduces the communication overhead of small draft models.
+
+        This is implemented by changing vLLM's tensor parallel group to a group
+        of the small size temporarily during forward passes of draft models.
+
+        Args:
+            draft_ranks (Optional[List[int]]): if this value is given, only some
+            of the GPU ranks in this value participate in draft generation
+        """
+
+        self._draft_ranks = draft_ranks
+        self._is_dummy = False
+        self._world_group = None
+        self._tp_group = None
+
+        if draft_ranks is not None:
+            # whether the worker participates in draft generation or not
+            self._is_dummy = kwargs['rank'] not in draft_ranks
+
+        super().__init__(**kwargs)
```

Comment:
Can we keep the super init invocation at the top? We want to avoid allowing any data dependency between this constructor and the superclass constructor if we can.